tools/fuzzer/afl-main.c

```diff
@@ -0,0 +1,498 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
+// Author: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <errno.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <pthread.h>
+#include <string.h>
+#include <signal.h>
+#include "fuzzer.h"
+#include <ipc/topology.h>
+#include <ipc/stream.h>
+#include <ipc/control.h>
+#include "qemu-bridge.h"
+#include <ipc/trace.h>
+
+/* set to 1 to enable debug */
+static int afl_debug;
+
+int enable_fuzzer;
+
+int ipc_reply_recd;
+pthread_cond_t ipc_cond = PTHREAD_COND_INITIALIZER;
+pthread_mutex_t ipc_mutex = PTHREAD_MUTEX_INITIALIZER;
+pthread_cond_t cond = PTHREAD_COND_INITIALIZER;
+
+/* tplg message types */
+uint32_t tplg_cmd_types[] = {SOF_IPC_TPLG_COMP_NEW,
+			     SOF_IPC_TPLG_COMP_FREE,
+			     SOF_IPC_TPLG_COMP_CONNECT,
+			     SOF_IPC_TPLG_PIPE_NEW,
+			     SOF_IPC_TPLG_PIPE_FREE,
+			     SOF_IPC_TPLG_PIPE_CONNECT,
+			     SOF_IPC_TPLG_PIPE_COMPLETE,
+			     SOF_IPC_TPLG_BUFFER_NEW,
+			     SOF_IPC_TPLG_BUFFER_FREE};
+
+/* PM message types */
+uint32_t pm_cmd_types[] = {SOF_IPC_PM_CTX_SAVE,
+			   SOF_IPC_PM_CTX_RESTORE,
+			   SOF_IPC_PM_CTX_SIZE,
+			   SOF_IPC_PM_CLK_SET,
+			   SOF_IPC_PM_CLK_GET,
+			   SOF_IPC_PM_CLK_REQ,
+			   SOF_IPC_PM_CORE_ENABLE};
+
+uint32_t comp_cmd_types[] = {SOF_IPC_COMP_SET_VALUE,
+			     SOF_IPC_COMP_GET_VALUE,
+			     SOF_IPC_COMP_SET_DATA,
+			     SOF_IPC_COMP_GET_DATA};
+
+uint32_t dai_cmd_types[] = {SOF_IPC_DAI_CONFIG, SOF_IPC_DAI_LOOPBACK};
+
+uint32_t stream_cmd_types[] = {SOF_IPC_STREAM_PCM_PARAMS,
+			       SOF_IPC_STREAM_PCM_PARAMS_REPLY,
+			       SOF_IPC_STREAM_PCM_FREE,
+			       SOF_IPC_STREAM_TRIG_START,
+			       SOF_IPC_STREAM_TRIG_STOP,
+			       SOF_IPC_STREAM_TRIG_PAUSE,
+			       SOF_IPC_STREAM_TRIG_RELEASE,
+			       SOF_IPC_STREAM_TRIG_DRAIN,
+			       SOF_IPC_STREAM_TRIG_XRUN,
+			       SOF_IPC_STREAM_POSITION,
+			       SOF_IPC_STREAM_VORBIS_PARAMS,
+			       SOF_IPC_STREAM_VORBIS_FREE};
+
+uint32_t trace_cmd_types[] = {SOF_IPC_TRACE_DMA_PARAMS,
+			      SOF_IPC_TRACE_DMA_POSITION};
+
+/* list of supported target platforms */
+static struct fuzz_platform *platform[] = {
```

Comment:
@ranj063 should another stretch goal be to also add the other non-cavs platforms?